react/react1/week3/homework/shift-app/src/usercomponents/UserList.js

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from "react";
+import FancyBorder from "./FancyBorder";
+import SubmitShift from "./SubmitShift";
+import SubTitle from "./SubTitle";
+import Search from "./Search";
+import User from "./User";
+import Total from "./Total";
+const UserList = ({ fetchUrl }) => {
+  const [users, setUsers] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setLoading(true);
+    const timeOut = setTimeout(getResult, 3000);
+    return () => clearTimeout(timeOut);
+  }, []);
+  const getResult = async () => {
+    setLoading(true);
+    try {
+      const response = await fetch(fetchUrl);
+      const result = await response.json();
+      setUsers(result);
+    } catch (err) {
+      setError(err);
+    }
+    setLoading(false);
+  };
+
+  const addEmployee = (user) => {
+    setUsers((prevValue) => {
+      return [...prevValue, user];
+    });
+  };
+
+  const handleFilter = (search) => {
+    setUsers((prevValues) =>
+      prevValues.filter((prev) =>
+        prev.name.toLowerCase().includes(search.toLowerCase())
+      )
+    );
+  };

```

Comment:
You can add getResult Function in handleFilter. Your solution is deleting other users from statement and we can search for only one time. With this solution you will set all users before search.
```suggestion
  const handleFilter = async (search) => {
    const users = await getResult()
    setUsers((prevValues) =>
      prevValues.filter((prev) =>
        prev.name.toLowerCase().includes(search.toLowerCase())
      )
    );
  };

```

Reply:
thank you so much , I  did not notice that and that was very interesting .